.github/workflows/test.yml

```diff
@@ -465,3 +471,42 @@ jobs:
           asset_path: gobra-ide/client/gobra-ide.vsix
           asset_name: gobra-ide.vsix
           asset_content_type: application/octet-stream
+
+      # compare version in client/package.json with last published version on
+      # VS Marketplace and deploy this version if newer.
+      # credits go to @fpoli!
+      - name: Obtain version information
+        run: |
+          VSCE_OUTPUT="$(
+            npx vsce show viper-admin.gobra-ide --json
+          )"
+          if [[ $(echo $VSCE_OUTPUT | grep --fixed-strings --line-regexp undefined) ]]; then
+            LAST_PUBLISHED_VERSION="0"
+          else
+            LAST_PUBLISHED_VERSION="$(
+              echo $VSCE_OUTPUT | jq '.versions[0].version' --raw-output
+            )"
+          fi
+          CURRENT_VERSION="$(
+            cat gobra-ide/client/package.json | jq '.version' --raw-output
+          )"
+          echo "LAST_PUBLISHED_VERSION=$LAST_PUBLISHED_VERSION" >> $GITHUB_ENV
+          echo "CURRENT_VERSION=$CURRENT_VERSION" >> $GITHUB_ENV
+
+      - name: Publish the extension to Visual Studio Marketplace
+        uses: HaaLeo/publish-vscode-extension@v0
+        if: env.CURRENT_VERSION != env.LAST_PUBLISHED_VERSION
```

Comment:
would it make sense to check that LAST_PUBLISHED_VERSION is earlier than CURRENT_VERSION instead of checking only that is different? Is it possible that a CI job can run with a version that is older than the one that is published and that it updates the published version by mistake to an older version?

Reply:
The version is a string with major, minor, and fix revision. Thus, checking for inequality is non-trivial. Furthermore, the marketplaces hopefully reject new releases that have an identical version number as a previous release. Last but not least, the version number in `package.json` has to be manually increased and there is some responsibility that a fresh version number is picked whenever the increase happens. I would rather not increase complexity in the CI scripts further